examples/caching_and_replay/README.md

```diff
@@ -0,0 +1,42 @@
+# Caching and Replaying Schelling
+
+## Summary
+
+This example applies caching on the Mesa [Schelling example](https://github.com/projectmesa/mesa-examples/tree/main/examples/Schelling).
+It enables a simulation run to be "cached" or in other words recorded. The recorded simulation run is persisted on the local file system and can be replayed at any later point.
+
+It uses the [Mesa-Replay](https://github.com/Logende/mesa-replay) library and puts the Schelling model inside a so-called `CacheableModel` wrapper that we name `CacheableSchelling`.
+From the user's perspective, the new model behaves the same way as the original Schelling model, but additionally supports caching.
+
+## Installation
+
+To install the dependencies use pip and the requirements.txt in this directory. e.g.
+
+```
+    $ pip install -r requirements.txt
+```
+
+## How to Run
+
+To run the model interactively, run ``mesa runserver`` in this directory. e.g.
+
+```
+    $ mesa runserver
+```
+
+Then open your browser to [http://127.0.0.1:8521/](http://127.0.0.1:8521/) and press Reset, then Run.
+
+After running a regular simulation, you can **replay** your latest simulation run by first enabling the Replay switch and then pressing Reset.
```

Comment:
Do I read this as:
- Run regular simulation with "Replay last run?" as False
- Then set "Replay last run?" to True
- Press "Reset"
- Start the simulation

What would happen if I set the replay switch to `True` in the beginning, while the cache is still nonexistent? Either throw an error, or implicitly run and save to cache. Either way, the user needs to be informed of the cache existence/non-existence.

Also, I think it should be called "Replay cached run?" instead. Because "last run" implies most recent run, which is not always the case.

Reply:
You understand the steps correctly. 

If the replay switch is set to `True` in the beginning, the user would get a "cache file not existing" error message. This would happen in the console though and not in the web view. Is there a simple way to inform the user about the cache file existence in the web view?

It would be okay for me to rename `Replay cached run` into `Replay cached run`. I think it should always be the case that this is the latest run though: when we simulate and do not replay, a new run is executed but will in any case be written to the cache. If there is a replay, it will replay the run from cache, which is the same as the latest simulation run or the latest replay.

Reply:
You can display custom information as a text, e.g. https://github.com/projectmesa/mesa-examples/blob/37e0a9de1e8c7ee9c907757d744053606539744c/examples/Schelling/server.py#L10. In this case, it would be to show whether the cache file exists.

> After running a regular simulation

You have to make it explicit that the user has to disable the switch for the first run.

>  If there is a replay, it will replay the run from cache, which is the same as the latest simulation run or the latest replay.

So, caching always happens whenever the server is not in replay mode. This needs to be explicitly told via a short explanation, possibly near the text info showing whether the cache exists or not. Anyway, "replay cached run" is still safer, because there is a possibility that the user will save the cache somewhere else, and to replay it on a different machine, etc.

Reply:
I've added explicit information about the caching status and that replay can be used only if a cache file already exists now.